### PR TITLE
attack animations

### PR DESCRIPTION
**Pull request by Intigracy**
https://dl.dropboxusercontent.com/s/3jjktbw71k4jnbh/aameeting.mp4?dl=0

http://ss13.moe/index.php/poll/51

:cl:
- tgs: Added attack animations, ported from TG.

### CODE REVIEW

**Comment by Probe1** [comment]
Fund it.  It's not ideal.  It isn't even good.  But I'm not looking at it as "we could either have this or something better".  We're either having this or nothing and for fucks sake, nothing is awful and leads to bad gameplay.  Players should not have to read a text box to see who attacked them while being attacked.

Anything is better than that.  And this isn't _that bad._

**Comment by kilozombie** [comment]
After enough ban appeals and actual bans and minute fuck-ups and wrongful arrests based off of assuming the wrong person is attacking somebody, this is pretty cool.
I guess I don't know the reasons why not to add something like this.

**Comment by Intigracy** [comment]
I made it half the movement amount as it is on TG because theirs is just insanely large.

**Comment by kilozombie** [comment]
Wait, that's not the animation I thought it was.
Isn't there another one that actually displays the item you're attacking with? That, at least, seemed to have more clarity

**Comment by SangriaMonkey** [comment]
:fu:

**Comment by Shadowmech88** [comment]
I agree with Probe. As it is, I don't think AIs even have a way to tell who's attacking who in a fight until one of them falls over, since they can't see emotes. Combat is generally a clusterfuck and having to read chat just to get a sense of what's happening adds to that.
It'd be really great if there was some other solution that solved those problems while being better than this, but if there is, I can't think of it.

Also, I think the animation should be faster. More like a bump than a slide towards the target.

**Comment by kilozombie** [comment]
Well, like I said, there are 'animations' that don't look dumb out there. Can't remember what codebase the ghost-item thing is, exactly, or if that looks stupider.

**Comment by Shadowmech88** [comment]
I imagine it would look worse unless they sprited an icon specifically for use with that animation, for every item.

**Comment by kilozombie** [comment]
No, what it does is use the item in hand that you're attacking with, and displays it attacking the person you're attacking, I think.

**Comment by Exxion** [comment]
I honestly couldn't care less what variety of attack animation we use as long as we get one

For reasons incomprehensible, the thread is already flipping its shit about this, but that was probably expected because for some reason they've always had some kind of phobia of attack animations

**Comment by Shadowmech88** [comment]
Because "TG has so many better features than us" yet if we port any TG features we're in danger of becoming TG.

**Comment by kilozombie** [comment]
People who say "TG has so many better features than us" aren't the same people that say "we shouldn't merge this TG thing because we're not TG". The thread isn't a hivemind.

I understand the phobia a little better now. With attack animations comes more fluid and obvious and dumb combat, which might lead to more greyshittery or deathmatch by proxy. Personally I don't think we're gonna slip on down that slippery slope, but look at Hippie or /tg/ if you want an idea for how combat starts and ends there.

**Comment by ghost** [comment]
@Intigracy 
>+1ing your own PR

**Comment by yclatious** [comment]
I always advocate the player paying attention, everything is obvious, combat in particular, you don't need to have attack animations to see who is fighting, never really did, the player facing people, running, the classic sound of a toolbox and a stun baton, its all obvious, if you want exact names, a quick glance at the chatbox, but for just about every situation, its clear what is happening.

Also, tg has a lot of features, some decent, some hot cancerous garbage shit. We don't want most of them because we don't just eat hot cancerous garbage like tg does, most features don't mesh well here, with both the code and playerbase.

**Comment by Exxion** [comment]
Requiring you to have a lazy eye so you can keep one eye on the game panel and one on the chatbox is artificial difficulty

**Comment by kilozombie** [comment]
Haven't played AI in a while but I don't think AI gets emotes in the chatbox, does it? Nor any sound. So, I mean, you're rolling off way less when observing combat.

**Comment by yclatious** [comment]
Being so distracted you cannot identify a obvious combat situation or be able to identify someone by the very clear red text saying very clearly who is doing what is plain outright feature padding for people who can't pay attention.

If you are a AI, you stun both, get them both to medical help. You can identify combat situations fine enough, if you play AI and can't pay attention to player behaviour or identify it, you shouldn't play AI, you won't see it flare up either since you are busy watching over everyone else, now having a animation telling you the obvious won't help.

**Comment by Shadowmech88** [comment]
>man falls over
>paramedic runs up to him, medkit in hand
>stands there for a few seconds

How, at this point, is the AI supposed to know whether the paramedic is treating his wounds or beating his head in with the medkit?

**Comment by Exxion** [comment]
"We don't need it" isn't an argument against having it. Just a LACK of argument FOR having it.
Plus you're fucking delusional if you think every single combat situation is obvious. You're right that it is usually clear for veteran players, but certainly not always, and almost NEVER for new players.

**Comment by kilozombie** [comment]
>If you are a AI, you stun both,

What in the fuck did he mean by this
I mean... are you okay? Do you need alkysine?

**Comment by TheFriggenSawbones** [comment]
it looks pretty ok

**Comment by yclatious** [comment]
Nobody beats heads with medkits shadow, a paramed would debrain him, lethal chloral or do something worse, nobody would be dumb enough to medkit someone to death. Was this meant to be a argument against me and my whole "knowing player behaviour" argument, you could even see the paramed become bloody if he did that anyways anyways.

Just about every single one of them is, yes,  new players are new, why should you tell them not to pay attention, have them not he brainless atentionless cockatoos, don't make a system just so the folks who barely master hands and still type Say"" can say greyshit mcobviousantag hit them.

Stun both so whoever is not stunned won't start curb stomping the other guy, going under assimov, you can't let inaction lead to harm, its safer to stun both so the situation is proper defused.

**Comment by kilozombie** [comment]
>If you are a AI, you stun both, 

Please yclat I need answers on this, I'm so confused

**Comment by yclatious** [comment]
4 hours of sleep, brain is mush right now, meant as your best orders to give to a borg, the best course of action.

**Comment by Exxion** [comment]
But what's your actual argument *against* it? How would it *hurt* gameplay to add this?

**Comment by JustSumBody** [comment]
Here's the way I see it:

Attack animations benefit new and even sometimes veteran players by allowing them to gain more information at a glance rather than having to look at chat or whatever.
Quality of life to AIs and I guess sec who can see fighting through cameras.

Unless you can give an actual DOWNSIDE to having them, then your being opposed to it is purely on the basis that you personally don't gain anything from it.

Or you're cluster and think the way it's implemented in the code is kind of shit, but I really don't think you're in a position or aware of how to make that argument.

If you don't have a solid argument AGAINST it(an argument that says "implementing this would cause problems because _____") then you're just making some shit "no true scotsman" argument.
And shit arguments are dumb.

**Comment by kilozombie** [comment]
Right.
I'm just not sure why forcing people to look in the chatbox for certain things is going to make them more robust. It'll teach them to read faster, maybe? I guess? I just see a lot of scenarios in the game becoming way less shitty with SOME form of visual clarification for attacks.
The downside is a possible detriment to the community. Sure, OK, argue that. I'm not a good judge on this, seeing as I'm a literally who from a different server.

**Comment by kilozombie** [comment]
Get that controversial tag in there motherfucker
Also remove quality, I mean... it's a direct-to-DVD port

**Comment by Exxion** [comment]
I'll add [Controversial] when more than one (1) person is arguing against it
And [QUALITY] means quality-of-life, of which this is a textbook example

**Comment by kilozombie** [comment]
Announce it in OOC tomorrow, get some people who aren't on at /real-dead-hours/.

**Comment by yclatious** [comment]
I sincerely think they looks awfull to begin with, and my main argument is the lazy mentality behind them exxiom. Its clear you folks want it and I'm not the majority here, but this just yells to me as a lazy crutch so people can now mindlessly just be told the already obvious, with no effort required. Its a shit mentality, from a shit server, who promoted  and promotes lazyness. Remember the timed antag bombs in that one PR we had, that everyone shat on, that's kind of my point, slippery slope or not.

Maybe what I'm preaching is wrong, and if it is, I'll gladly accept it, but I don't see this actually doing much to help to begin with like I mentioned before, its a feature that incentivises lazyness by its design, and while it may be more user friedly, do we sincerely want or need to cater to folks who can't pay attention to anything.

If anything, people arguing about AI not being able to tell people attacking of getting attack logs and such are right and deserve a fix to that instead of this, since this will at best be a bandaid fix. Get them the ability to see any fighting battle logs when in sight, not a feature about seeing who shot first when its a realistic expectation that AIs wont see every single starting fight, hell, Id argue the confusion in that is good.

**Comment by SangriaMonkey** [comment]
Dry-humping your opponent to death looks fucking silly, sorry.
The variant that uses the item's icon is a little less aggravating.

**Comment by Exxion** [comment]
Again, you're right, this is not necessary in most circumstances
But occasionally it is (well not necessary, but very helpful), and it's always somewhat helpful

And yeah, it does look kind of silly, but it's better than nothing
Anyone is free to port the other variety instead of this one
Though, judging by the cursory glance I gave this code, it would probably be easier to instead change it to the other one after this is merged

**Comment by clusterfack** [comment]
I greatly greatly dislike the code for this, as evidenced by the fact I didn't open it myself. Putting it spaghetti throughout the code makes this extremely unmaintainable.

**Comment by Exxion** [comment]
Wait actually yeah why IS it like that?

File: code/game/machinery/shieldgen.dm [hunk 1]
```diff
@@ -57,15 +58,15 @@
 
 
 	if (src.health <= 0)
-		visible_message("<span class='notice'>The [src] dissapates</span>")
+		visible_message("<span class='notice'>The [src] dissipates</span>")
```
Comment (Exxion) [inline code/game/machinery/shieldgen.dm]: UNATOMIC, etc.

**Comment by yclatious** [comment]
Not to be a prick, but settling for dirt when you can have riches seema silly, don't we have niggles and sprinters who could atleast cook up a decent animation?

If you folks are going to think its not at all lazy design and made with a mentality of lazy behaviour, least you can do is yourselves not be lazy and improve it, otherwise you are just making poetry at this point.

**Comment by Exxion** [comment]
What would you rather it looked like, in a general sense?

**Comment by JustSumBody** [comment]
> Get them the ability to see any fighting battle logs when in sight

Reminder that I tried this and it got shut down as POWER CREEP

Also you're literally arguing "user friendliness" makes people lazy and making yourself out to be some HARDCORE GAMER who's so ULTRA AWARE OF EVERYTHING. 
Again, no true Scotsman is a shit argument and you shouldn't be using it.

**Comment by SangriaMonkey** [comment]
@Exxion 
The item's icon, slightly opaque, moving towards the targeted person or thing, as opposed to people bouncing off of corpses and windows repeatedly.

**Comment by yclatious** [comment]
Long as it isn't ghost weaponry or anything but slightly walking forward half a tile, it should look good exxiom. Goons animations looks clear, if you wanted to do a animation, that'd be a good idea for it. Or just steal goon if you don't want to make a new one or just can't be assed.

I'm a regular player Body, I dont claim to be the best, not at all, plenty of folks are better and pay a lot more attention, I just pay a lil attention and see shit clearly, hence my confusion at seeing you folks which I am dang sure are better than me, needing/wanting this.tg is lazy as shit though, let's be real.

A shame it was called power creep, seems fine to me and a boon for AI players, I'd support it tbh.

**Comment by kilozombie** [comment]
Goon's animation looks ok. It's really just the sprite jiggling around or something?

**Comment by yclatious** [comment]
Pretty much, yeah.

**Comment by kilozombie** [comment]
For me, it's correlation vs causation
Every server I know of with attack anims has a lot of shittery and deathmatch, but I don't know if it's a partial result of the attack anims or just general server culture.

**Comment by SangriaMonkey** [comment]
Those servers rely on reductive design and are quick to baby-proof things before too many feelings are hurt. We should strive to be different.

**Comment by Exxion** [comment]
I get the whole "keep it hardcore" sentiment, but at this point not having attack animations is just bad game design
Them looking bad is a valid (if minor) complaint, though
Again, I don't personally care what the attack animations look like as long as we get them
If everyone agrees that the Goon (?) animations are better, maybe we should port them instead.

Added bonus: Those ones could theoretically be made toggleable on a per-client basis, unlike these ones

**Comment by SangriaMonkey** [comment]
A toggle for them would be great.

**Comment by yclatious** [comment]
Make them toggleable and client sized and I will have no gripes with them, by all means, you folks enable em, I'll keep em off.

**Comment by optimumtact** [comment]
Let the madness of the tg warp wash over you brothers

**Comment by yclatious** [comment]
They made up their minds citrus, can't change it now

**Comment by Ephx1** [comment]
I don't really care but didn't pomf specifically not want this for whatever reason?

**Comment by clusterfack** [comment]
If these images were set to a particular plane they could easily be hidden from everyone at once from a simple preferences toggle that makes the plane master set them to zero alpha.

**Comment by ghost** [comment]
They're not images but animations though

**Comment by ihadtoregisterforthis** [comment]
This looks incredibly bad. I cannot think of a worse way of displaying attacks visually.

Even comical bat-fight *POW* and *BIFF* would look better than this. Even a floating damage number would look better than this. Even the victim blinking red when hit would look better than this. Even small particles falling off would look better than this.
The attacking item appearing over the victim would be better than this if things like mobs were accounted for too.

The fact that this exists says a lot for the server this comes from.

**Comment by ghost** [comment]
I don't mind the animations, I guess. I don't really like them either, but the one thing I genuinely dislike is the fact there's an animation when point-blank shooting someone. That just makes no sense. 
Other than that, it just looks really, really silly. 
I know lots of people have brought up the point that 'you have to look in your chatlog to see who attacked you' but, unless you're sitting in a 40 inch TV trying to place space, can you not figure out a way to balance your screen so you can encompass everything? 
I mean, you are playing a game where there's only text chat, and everything only comes through the text log. Surely, you should be focusing on both at once so you don't miss anything important.

**Comment by ghost** [comment]
https://github.com/ParadiseSS13/Paradise/pull/4243
https://github.com/tgstation/tgstation/pull/10314
https://github.com/tgstation/tgstation/pull/13298

The ghost item animations

**Comment by yclatious** [comment]
Just because two shit to mediocre stations did it doesn't mean its good uranium.

**Comment by ghost** [comment]
https://github.com/ParadiseSS13/Paradise/pull/4274

Preference for the ghost item thing

**Comment by SonixApache** [comment]
I like how silly it is.

**Comment by Allspice** [comment]
Didn't we have this sort of "out of place" objection for action progress bars

**Comment by 9600bauds** [comment]
It's great, but why is the attack animation playing for point blank gunshots? It's not a pistolwhip or something

**Comment by ComicIronic** [comment]
Yes, but progress bars are toggleable. As currently implemented, this would not be.

Where is the server vote on having any kind of animation at all?

**Comment by 9600bauds** [comment]
See OP:
http://ss13.moe/index.php/poll/51

**Comment by kilozombie** [comment]
I hate votes like that.
Out of the 74 people that voted, 33 will be unhappy with the change.
That, in itself, is enough to make this 'controversial'.

**Comment by Probe1** [comment]
@ComicIronic in the OP of the PR.

**Comment by Intigracy** [comment]
That's not a point blank gunshot ffs it's hitting them with the gun.

File: code/game/machinery/shieldgen.dm (hunk 1 above)
Comment (Intigracy) [reply code/game/machinery/shieldgen.dm]: If you're actually serious then I'll revert it, DrCelt.

**Comment by 9600bauds** [comment]
Harm intent gun melee = pointblank gunshot
Non-harm intent gun melee = hitting them with the gun
In the webm, you were in harm intent

File: code/game/machinery/shieldgen.dm (hunk 1 above)
Comment (ComicIronic) [reply code/game/machinery/shieldgen.dm]: I'm serious. These changes aren't relevant to the PR, and we shouldn't have to roll back fixes as part of content changes.

Comment (9600bauds) [reply code/game/machinery/shieldgen.dm]: Note that when people make these comments they don't mean it, it's just memes

**Comment by Intigracy** [comment]
I'm not seeing a way to fix that specifically since attackby is called before attack

**Comment by ghost** [comment]
@yclatious 
>Just because two shit to mediocre stations did it doesn't mean its good uranium

Still better solution to identify attackers than this PR has. It may look silly but make game more clear and user-friendly

**Comment by yclatious** [comment]
Ive already said how its lazy as shit, used by lazy as shit servers, with lazy as shit players, and I stand by it, if ss13 wanted to be user friendly by design, it wouldn't have been build the way it currently is.

Better solution, but still not a good one. Make it toggleable and client sided and ill accept you folks having your crutch, just don't force it on me if you intend to casualize the detection of what already takes almost no effort to notice or recognise.

**Comment by optimumtact** [comment]
hahah yeah that's it bro, keep on being non user friendly, having only 20 players at prime time is good for you!

**Comment by yclatious** [comment]
We had so many players recently, Pomf actually budged and triggered the bunker after months of inactivity citrus. 

Plus, better small and with 40 regular max pop than whatever clusterfuck TG, paradise and CM are. Less players but better ones seems fine as hell to me, maybe you got lower standards, but I sure don't and don't need to lower em just so people who cant pay minimal attention can be spoonfed information they already could tell by the least amount of effort possible being put into the game.

**Comment by kilozombie** [comment]
Have it default off and be obscured away in preferences somewhere, then, if you want new players to get used to not having it. Not the worst option, really.

**Comment by SonixApache** [comment]
Animations would be a lot user friendlier.
Opt-out seems like it'd be a good option for the hardcore text purists among us.

**Comment by d3athrow** [comment]
my opinion is the same as always, the tg attack animations look like ass and detract from the gameplay but what does my opinion matter anymore?
Needs to be toggleable and by default off or it will just foster a new era of casuals who have no situational awareness because they're babied by half baked animations.

**Comment by 9600bauds** [comment]
Has any other server figured out how to make client-toggleable attack animations? Especially the non-"itemghost" kind

**Comment by Exxion** [comment]
This kind cannot be feasibly made toggleable.
The itemghost kind can easily be made toggleable

**Comment by ComicIronic** [comment]
why don't we flatten the mob icon every attack and send the image to the client for toggleability

**Comment by 9600bauds** [comment]
Because then we'd be flattening the mob icon every attack and sending the image to the client

**Comment by Exxion** [comment]
>feasibly

**Comment by clusterfack** [comment]
I already said how to make them invisible client side, plane masters

**Comment by Probe1** [comment]
>Pomf saying the same thing I said about progress bars
You know how I felt.

**Comment by kilozombie** [comment]
I think it should start default off.
Reasoning: a new player in combat will benefit from having to check the text box. Not because reading redtext is particularly fun or easy, but because a new player should learn the habit of checking it, for people shouting or performing actions, etc.
It's a shitty solution, but most people in here learned without attack anims, right? Personally, having to check the text box taught me by proxy to be mindful for people saying things, the radio, etc.
Otherwise you get people who click on spessmen with their toolbox and never, ever check the chat, which is absolutely fucking awful.

**Comment by sonime** [comment]
If it's going to be toggleable, make the default off so we can identify the TGfags.

**Comment by Exxion** [comment]
How would you identify someone from a clientside toggle

**Comment by 9600bauds** [comment]
People said a lot of shit about progress bars and virtually everyone has them enabled nowadays

**Comment by sonime** [comment]
If they mention it or make a webm etc

**Comment by Exxion** [comment]
It sounds like the consensus (including Pomf) is to get the ghost item version instead of this so that it looks better and/or is toggleable

**Comment by 9600bauds** [comment]
I don't see a consensus for the ghostitem version

**Comment by Exxion** [comment]
Pretty much every complaint has been that this looks bad and the only alternative suggested has been that
Also, that's the kind that could be made toggleable
A sprite other than the ghost item could also be toggleable, but this kind specifically cannot

**Comment by Probe1** [comment]
ghost items are pretty ghetto looking and manage to look worse than attack anims alone to me.

poll
pr

all want one thing, and that thing is something people have asked for going on a year and a half, and most other servers have already adopted.  

don't be scared of change.  this isn't a big deal.

**Comment by Exxion** [comment]
Actually you COULD make this kind toggleable *slightly* less awfully than @ComicIronic said
You wouldn't have to flatten the person every time. You could just assign the image's `appearance` to the attacker.
It would still be pretty awful though, requiring two images to be kept per person and sent to every client logged on and updated with the new`appearance` every attack.
Essentially any other possibility would be better if you want it toggleable.

**Comment by optimumtact** [comment]
Also I just want to discuss one more thing
> if ss13 wanted to be user friendly by design, it wouldn't have been build the way it currently is.

That's a stupid thing to say, it was built the way it was because that's all people had the time or will to do, not because of some design goal to make it unintuitive on purpose.

**Comment by 9600bauds** [comment]
Or the ability to do, how many things do we still not have solely because of BYOND?

**Comment by Intigracy** [comment]
Where has Pomf said anything about this?

**Comment by ihadtoregisterforthis** [comment]
Here, four hours ago, before labeling controversial.

**Comment by Intigracy** [comment]
Yeah, I'm retarded.

**Comment by Probe1** [comment]
oh don't be that way

the majority wants it.

**Comment by Exxion** [comment]
Again, I'd say merge this then change it to a different type later if people want that

**Comment by 9600bauds** [comment]
We've "test" merged far worse things.
I'd say put this in for 3 days/1 week, hold another server vote to see what the reception is like, revert it unless it gets at least 75% positive votes.

**Comment by kol1th** [comment]
How is this controversial again

**Comment by Intigracy** [comment]
Soon to remove HUD and replace it with text commands to check/change things around in your inventory.

**Comment by Probe1** [comment]
Because 30% dont want it and thats fair to say when 30% are quite a few people.

**Comment by Iamgoofball** [comment]
@Probe1 that means 70% wants it you dumbfuck

**Comment by Intigracy** [comment]
@Iamgoofball I only see one dumbfuck here and his name is a pronoun, a verb, and then an adjective.

Since you don't have any brains, I'll spell it out for you:
It's you.

**Comment by Iamgoofball** [comment]
how does that dispell the whole "70% of people want it" friend

**Comment by 9600bauds** [comment]
Goofball no goofing

**Comment by Intigracy** [comment]
Goofball no goofing

**Comment by Iamgoofball** [comment]
Goofball no goofing

**Comment by Iamgoofball** [comment]
real meme talk though: attack animations improved the shit out of the game at tg, it cut down confusion about who's attacking who by almost half, just ask our admin team

**Comment by kol1th** [comment]
It's literally just an improvement but people will scream because it's not what they are used to and it's "TOO CASUAL"

**Comment by kol1th** [comment]
also Goofball no goofing

**Comment by Intigracy** [comment]
https://dl.dropboxusercontent.com/s/3jjktbw71k4jnbh/aameeting.mp4?dl=0

**Comment by kol1th** [comment]
perfect

**Comment by Intigracy** [comment]
This still has some finishing touches.

**Comment by Intigracy** [comment]
Though that should be the last of them, unless anyone else sees anything.

**Comment by kilozombie** [comment]
Toggleable?

**Comment by ghost** [comment]
let me toggle it off and it's fine

**Comment by Iamgoofball** [comment]
@kilozombie @solariousgabriella

Whiny cunts, stop bitching about ways to turn everything ypur autistic minds refuse to immediately adjust to off

Try getting used to something new for once in your lives and you might be able to move out of your parents basement

**Comment by Kammerjunk** [comment]
Goofball no goofing

**Comment by 9600bauds** [comment]
We'd love to "just" make it toggleable.

**Comment by kilozombie** [comment]
Are you okay, my guy?
I don't give a shit whether or not attack animations are off. I'd like to have them on for my game. I've got reason to think having it toggleable would make a lot more people happy and be better for the game.

**Comment by 9600bauds** [comment]
Yeah, I was saying that them being toggleable would be a good thing for everyone, obviously
But it's not that simple

**Comment by kilozombie** [comment]
Ah, I meant that for Goof.
If it's not feasible, oh well. Some people will be upset with it being merged, but I guess that's nothing new. 
@Iamgoofball 
There's no need to antagonize people who dislike a change just because you think you're absolutely right-- it's always worth listening to the opposite argument if you actually give a shit about being reasonable. For instance, screeching at me like a retard for simply asking if the feature would be toggleable.

**Comment by Ephx1** [comment]
Might wanna add Goofball's entire name, you just pinged a completely random person who works on iOS stuff.

**Comment by kilozombie** [comment]
Poor goof.
Sorry goof.

**Comment by Intigracy** [comment]
I'll re-examine whether this is worth keeping/adjusting in a week.

The keep/revert/change poll will get started on the 14th.

**Comment by ihadtoregisterforthis** [comment]
Sure, why discuss it with anyone else in the first place, you can always re-examine it yourself.

**Comment by aceedex** [comment]
There are not enough ways for me to express my disgust so I'll simply say

Fuck off back to tg

**Comment by kilozombie** [comment]
Flat-out ignoring pomf and self-merging a controversial change is a really, really shitty thing to do. You couldn't have waited until somebody came up with a better animation, a toggle, anything?
I'll be surprised if you ever 're-examine' this.

edit: Also, hahahaha, can't believe pomf was the one who had to add the Controversial tag even though it was clear from the start that people would dislike the change. Really, really shitty github ethic.

**Comment by SysadminJeroen** [comment]
Since the revert was done and you deleted the branch, will you make another PR or re-open this one?

**Comment by Gef21** [comment]
http://www.strawpoll.me/11815096

**Comment by Intigracy** [comment]
@Jeroen52 no

**Comment by 9600bauds** [comment]
Why are you making a strawpoll when we already had two other votes casted

**Comment by Gef21** [comment]
why not tbh

**Comment by 9600bauds** [comment]
You don't think at this point the vote may be a little biased by recent events?

I mean there has already been plenty discussion and that discussion culminated in the votes in the OP

**Comment by kol1th** [comment]
Why can't we have nice things

**Comment by kilozombie** [comment]
Vote right now seems pointless, and lol strawpoll of all things
The majority already wants it in in some form. Not really debatable.

**Comment by kol1th** [comment]
Glad there's a vocal minority that decides the fate of any PR they don't like

**Comment by ComicIronic** [comment]
>why can't we have nice things

Because the reversion was fundamentally about the self-merge and not the content?

**Comment by kilozombie** [comment]
@kol1th 
Glad there's reasonable people who know how Github (should?) work and have reverted a self-merge for a two-day-old controversial change so that it can be revised and merged later by somebody who actually has the authority to do it.
Literally the person that reverted it (PJB) has not stated his opinion, whether or not he likes the change.

**Comment by 9600bauds** [comment]
We need to have a test period for this, 3 days at least. And there needs to be another vote at the end of it. That would be the fairest for everyone.

**Comment by kilozombie** [comment]
If people are responsible with testing, and there's actually a vote, I'm completely behind that.

**Comment by 9600bauds** [comment]
We've had a real stupid history of "test" merges in the past, I take it upon myself to revert everything if there's not enough positive votes or if a vote doesn't get made.

**Comment by kilozombie** [comment]
Fuck yeah.

**Comment by Iamgoofball** [comment]
Just rig up some git magic for it

**Comment by Exxion** [comment]
OK I support attack animations but WHY the FUCK did you think self-merging this was an even REMOTELY appropriate idea?

**Comment by Probe1** [comment]
Merge it then @Exxion 

🍯

**Comment by Exxion** [comment]
Too late

**Comment by Intigracy** [comment]
I don't care enough about any of this to continue trying with it.

Though I'm not sorry I merged it, it was going to sit here another month or two while people parroted the same exact arguments back and forth.

If you care enough to add them yourselves, the easiest way to do it is to open a revert on the revert.

**Comment by Probe1** [comment]
Just re-open the PR and hit unsubscribe then.

**Comment by Iamgoofball** [comment]
@Probe1 i don't think you get what he means here

**Pull request merged**